apps/mobile/src/features/ImportPrivateKey/components/LoadingImport/LoadingImport.container.tsx

```diff
@@ -0,0 +1,56 @@
+import { selectActiveSafe } from '@/src/store/activeSafeSlice'
+import { useAppDispatch, useAppSelector } from '@/src/store/hooks'
+import { useSafesGetSafeV1Query } from '@safe-global/store/gateway/AUTO_GENERATED/safes'
+import { useEffect } from 'react'
+import { LoadingImportComponent } from './LoadingImport'
+import { useLocalSearchParams, useRouter } from 'expo-router'
+import { addSigner } from '@/src/store/signersSlice'
+
+export function LoadingImport() {
+  const { address } = useLocalSearchParams()
+  const dispatch = useAppDispatch()
+  const router = useRouter()
+
+  const activeSafe = useAppSelector(selectActiveSafe)
+  const { data } = useSafesGetSafeV1Query({
+    safeAddress: activeSafe.address,
+    chainId: activeSafe.chainId,
+  })
```

Comment:
What happens if that call fails with a network error? And why do we need to do a network request here? How did we get to the import page? We start from a safe, so we most probably have the safe in the store, no?

Reply:
No, because if the safe is added / removed during the time the user is importing it, it will not reflect here. So using rtk here will first give us the cached data and then will return the updated safe list, so there is no needing for adding it in the redux.

Reply:
How can a safe be added/removed during the time the user is importing a signer? 

I'm fine with leaving the code as is, I just don't understand how the above could happen?

Reply:
It can happen in the following scenario:

Suppose it is a safe with some owners and small threshold

- User opens the import signer screen but takes longer to import the private key, maybe he went to the bath or something 😆 
- One of the signers removed the signer the user wanted to add there
- Suppose the other signers were faster than the app user to approve this remove signer transaction
- The signer is removed
- User is still on this screen, and when he tries to add the signer, the app would add the signer however the signer is not part in the safe anymore, and since we list the signers in the signer list, the user would have "you do not have signers imported" message again, what would be weird.

But I agree with you, it might not happen though. Wdyt?